server/websocket.go

```diff
@@ -955,9 +958,60 @@ func (s *WebsocketServer) onNewBlockAsync(hash string, height uint32) {
 	glog.Info("broadcasting new block ", height, " ", hash, " to ", len(s.newBlockSubscriptions), " channels")
 }
 
+func (s *WebsocketServer) publishNewBlockTxsByAddr(block *bchain.Block) {
+	s.addressSubscriptionsLock.Lock()
+	publishNewBlockTxs := func() bool {
```

Comment:
Why is this a function? So that you can break the inner loop by return? 
I am a bit worried that it could be fairly slow if there are many channels with many subscribed addresses and only few of them subscribe to publishNewBlockTxs and this code is executed for every block. How about having a counter of publishNewBlockTxs subscriptions, when on subscribe we increment the counter and on unsubscribe (or on channel close) we decrement the counter. Then on new block the condition is simple (and we could do it even before starting the goroutine).

Reply:
68d930ae98cec403808d5aaa3e0f1704c8b6e628